libslide/src/scanner.rs

```diff
@@ -0,0 +1,209 @@
+mod types;
+use types::TokenType;
+use types::Token;
+
+pub struct Scanner {
+    input: String, 
+    pub output: Vec<Token>
+}
+
+impl Scanner {
+    // instantiate a new scanner
+    pub fn new<T: Into<String>>(input: T) -> Scanner {
+        Scanner{
+            input: input.into(),
+            output: Vec::new()
+        }
+    }
+
+    // matches token with symbol and creates it: private helper function 
+   fn create_symbol_token(c: char) -> Token{
+        let t = match c {
+            '+' => TokenType::Plus,
+            '-' => TokenType::Minus,
+            '*' => TokenType::Mult,
+            '/' => TokenType::Div,
+            '%' => TokenType::Mod,
+            '^' => TokenType::Exp,
+            '=' => TokenType::Equal,
+            '(' => TokenType::OpenParen, 
+            ')' => TokenType::CloseParen, 
+            '[' => TokenType::OpenBracket,
+            ']' => TokenType::CloseBracket,
+             _  => TokenType::Invalid(c.to_string())
+        };
+        let ret = Token{token: t};
+        return ret;
```

Comment:
```suggestion
        Token {token: t}
```

Let's also rename the `token` field to `type`.

Reply:
type is a keyword. lets use token_type